Assets/MixedRealityToolkit.ThirdParty/MRTK-Quest/Scripts/Input/Controllers/OculusQuestHand.cs

```diff
@@ -337,10 +341,30 @@ protected bool UpdateHandData(OVRHand ovrHand, OVRSkeleton ovrSkeleton)
                 }
             }
 
+            // Pinch was also used as grab, we want to allow hand-curl grab not just pinch.
+            // Determine pinch and grab separately
+            MixedRealityPose wristPose, indexKnucklePose, indexTipPose;
+            if (jointPoses.TryGetValue(TrackedHandJoint.Wrist, out wristPose))
+            {
+                if (jointPoses.TryGetValue(TrackedHandJoint.IndexTip, out indexTipPose))
+                {
+                    if (jointPoses.TryGetValue(TrackedHandJoint.IndexKnuckle, out indexKnucklePose))
+                    {
+                        // compare wrist-knuckle to wrist-tip
+                        float wristToIndexTip = Vector3.Distance(wristPose.Position, indexTipPose.Position);
+                        float wristToIndexKnuckle = Vector3.Distance(wristPose.Position, indexKnucklePose.Position);
+                        bool grip = wristToIndexKnuckle >= wristToIndexTip;
+
+                        IsGrabbing = grip;
+                    }
+                }
+            }
+
             if (MRTKOculusConfig.Instance.UpdateMaterialPinchStrengthValue && handMaterial != null)
             {
-                handMaterial.SetFloat(pinchStrengthProp, pinchStrength);
+                handMaterial.SetFloat(pinchStrengthProp, IsGrabbing ? 1.0f : pinchStrength);
```

Comment:
Id prefer if another float was used above that performed this grab check to not add logic in this material setter. Just a minor nit however.